### PR TITLE
Initial support for the status endpoint

### PR DESCRIPTION
**Pull request by CJLove**
Comments/suggestions welcome; I've tried to implement support for the status endpoint similar to how you support other endpoints...

Tested with consul 1.0.1 and 1.0.6 on mac and SLES 12.

### CODE REVIEW

File: include/ppconsul/status.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+namespace ppconsul { namespace status {
+
+	namespace impl {
+		std::string parseLeader(const std::string&);
+
+		std::vector<std::string> parsePeers(const std::string&);
+	}
+
+	class Status 
+	{
```
Comment (oliora) [inline include/ppconsul/status.h]: Put vsibility keywords on the same level as class (check existing code)

File: src/status.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+#include "ppconsul/status.h"
+#include "s11n_types.h"
+
+namespace ppconsul { namespace status {
+
+namespace impl {
+
+	std::string parseLeader(const std::string &json)
```
Comment (oliora) [inline src/status.cpp]: Coul you please put reference to the type name `const std::string& json` to match existing code.

File: include/ppconsul/status.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		public:
+		explicit Status(Consul &consul)
+		: m_consul(consul)
+		{}
+
+		std::string leader() const
```
Comment (oliora) [inline include/ppconsul/status.h]: Since leader is optional, let’s return it as `boost::optional<std::string>`.

**Comment by CJLove** [comment]
Code formatting cleaned up as requested.  Changed Status::leader() to use boost::optional

File: include/ppconsul/status.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        boost::optional<std::string> leader() const
+        {
+            auto leader = impl::parseLeader(m_consul.get("/v1/status/leader"));
+            return boost::optional<std::string>((leader != ""),leader);
+        }
+
+        bool isLeaderElected() const
+        {
+            return leader().is_initialized();
```
Comment (oliora) [inline include/ppconsul/status.h]: `leader() ? true : false`

File: include/ppconsul/status.h [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    namespace impl {
+        std::string parseLeader(const std::string&);
+
+        std::vector<std::string> parsePeers(const std::string&);
+    }
+
+    class Status 
+    {
+    public:
+        explicit Status(Consul &consul)
```
Comment (oliora) [inline include/ppconsul/status.h]: Please move ampersand to the left.

**Pull request merged**